pyfr/integrators/base.py

```diff
@@ -130,6 +134,32 @@ def cfgmeta(self):
         else:
             return {'config': cfg, 'config-0': cfg}
 
+    def grad_pvars(self):
+        try:
+            # Fetch the corrected gradients only if needed
+            grad_soln = self.grad_soln
+        except NotImplementedError:
```

Comment:
No need to try/catch; instead we can stick a better message in when we raise the NotImplementedError.

Reply:
Do you prefer raising an error in the NotImplementedError then?

Reply:
Yes, I think that is the most logical place.